bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -631,12 +631,13 @@ class PlotCanvasView extends Renderer.View
   _map_hook: (ctx, frame_box) ->
 
   _paint_empty: (ctx, frame_box) ->
-    @visuals.border_fill.set_value(ctx)
-    ctx.fillRect(0, 0,  @canvas_view.mget('width'), @canvas_view.mget('height'))
-    ctx.clearRect(frame_box...)
-    @visuals.background_fill.set_value(ctx)
-    ctx.fillRect(frame_box...)
-
+    ctx.clearRect(0, 0,  @canvas_view.mget('width'), @canvas_view.mget('height'))
+    if @visuals.border_fill.doit
+      @visuals.border_fill.set_value(ctx)
+      ctx.fillRect(0, 0,  @canvas_view.mget('width'), @canvas_view.mget('height'))
+    if @visuals.background_fill.doit
```

Comment:
Is there anywhere else besides `border_fill_color`? If not, not-repeating that is easy to do if that's the best choice, but let's make an actual decision one way or the other.